src/Microsoft.AspNet.Http.Features/IRequestCookies.cs

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.AspNet.Http
+{
+    /// <summary>
+    /// Represents the HttpRequest cookie collection
+    /// </summary>
+    public interface IRequestCookies : IDictionary<string, StringValues>
```

Comment:
Ahhh... remember now; wanted to return a strong type from the utility types e.g. `ParseQuery` and `ReadFormAsync` to avoid boxing on enumerator but `Microsoft.AspNet.Http` is after `Microsoft.AspNet.WebUtilities` in the build order; hence returning `KeyValueAccumulator` as a get around...